### PR TITLE
Allow TLV for single resource (use it as default format)

### PR DESCRIPTION
**Pull request by sbernard31**
The recent versions of the spec say now TLV can be used for single resources.

> "The LWM2M Client MUST support the TLV data format for all
requests." (§6.3 Data Formats for Transferring Resource Information)

This PR implements this modification and made the TLV format as default format in Leshan.

**Warning:** "Leshan Client" and "Leshan Server Demo" will use TLV format for all requests.

### CODE REVIEW

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/model/StandardModelProvider.java [hunk 1]
```diff
@@ -15,21 +15,11 @@
...
 package org.eclipse.leshan.server.model;
 
-import org.eclipse.leshan.core.model.LwM2mModel;
 import org.eclipse.leshan.core.model.ObjectLoader;
-import org.eclipse.leshan.server.client.Client;
 
-public class StandardModelProvider implements LwM2mModelProvider {
-
-    private final LwM2mModel model;
+public class StandardModelProvider extends StaticModelProvider {
```
Comment (jvermillard) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/model/StandardModelProvider.java]: JavaDoc?

**Comment by jvermillard** [comment]
LGTM

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/BootstrapWriteRequest.java [hunk 2]
```diff
@@ -59,9 +60,18 @@ public BootstrapWriteRequest(final LwM2mPath target, final LwM2mNode node, Conte
...
                         String.format("%s format must be used only for single resources", format.toString()));
             } else {
-                if (((LwM2mResource) node).isMultiInstances()) {
+                LwM2mResource resource = (LwM2mResource) node;
+                if (resource.isMultiInstances()) {
                     throw new IllegalArgumentException(
                             String.format("%s format must be used only for single resources", format.toString()));
+                } else {
+                    if (resource.getType() == Type.OPAQUE && format == ContentFormat.TEXT) {
+                        throw new IllegalArgumentException(String.format(
```
Comment (msangoi) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/BootstrapWriteRequest.java]: Nothing to format here :)

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java [hunk 3]
```diff
@@ -289,9 +289,18 @@ private WriteRequest(final Mode mode, ContentFormat format, final LwM2mPath targ
                 throw new IllegalArgumentException(
                         String.format("%s format must be used only for single resources", format.toString()));
             } else {
-                if (((LwM2mResource) node).isMultiInstances()) {
+                LwM2mResource resource = (LwM2mResource) node;
+                if (resource.isMultiInstances()) {
                     throw new IllegalArgumentException(
                             String.format("%s format must be used only for single resources", format.toString()));
```
Comment (msangoi) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java]: Nothing to format here

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java [hunk 4]
```diff
@@ -289,9 +289,18 @@ private WriteRequest(final Mode mode, ContentFormat format, final LwM2mPath targ
...
+                } else {
+                    if (resource.getType() == Type.OPAQUE && format == ContentFormat.TEXT) {
+                        throw new IllegalArgumentException(String.format(
```
Comment (msangoi) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java]: Nothing to format here

**Pull request merged**